### PR TITLE
@shopify/storybook-a11y-test

### PR DESCRIPTION
**Pull request by alex-page**
## Description

In `shopify/web` and `shopify/polaris-react` we have duplicated accessibility tests for storybook `/scripts/accessibility-check.js`. This package creates a single source of truth for the accessibility test moving forwards and allows other teams to use and enhance it from one place.

## Type of change

New package.

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version)

### CODE REVIEW

File: packages/storybook-a11y-test/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  const results = await storybookA11yTest(options);
+
+  if (results.length) {
+    console.error(chalk.bold(`‼️  Test failures found`));
+    console.log(messages.join('\n'));
+    process.exit(1);
+  } else {
+    console.log(chalk.bold('🧚‍♀️ Accessibility is all g'));
+  }
+});
```
Comment (alex-page) [inline packages/storybook-a11y-test/README.md:35]:
```suggestion
})();
```

File: packages/storybook-a11y-test/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  const results = await storybookA11yTest(options);
+
+  if (results.length) {
+    console.error(chalk.bold(`‼️  Test failures found`));
+    console.log(messages.join('\n'));
+    process.exit(1);
+  } else {
+    console.log(chalk.bold('🧚‍♀️ Accessibility is all g'));
+  }
```
Comment (alex-page) [inline packages/storybook-a11y-test/README.md:34]:
```suggestion
  if (results.length) {
    console.error(`‼️  Test failures found`);
    console.log(results.join('\n'));
    process.exit(1);
  } else {
    console.log('🧚‍♀️ Accessibility is all g');
  }
```

File: packages/storybook-a11y-test/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+// https://bugs.chromium.org/p/chromium/issues/detail?id=468153
+// This is necessary to prevent autocomplete in chrome and fails the axe test
+// Do not disable accessibility tests if you notice a failure please fix the issue
+const isAutcompleteNope = violation => {
+  const isAutocompleteAttribute = violation.id === 'autocomplete-valid';
+  const hasNope = violation.nodes.every(node =>
+    node.html.includes('autocomplete="nope"'),
+  );
+  return isAutocompleteAttribute && hasNope;
+};
```
Comment (alex-page) [inline packages/storybook-a11y-test/src/index.ts:75]: Ignore `autocomplete="nope"`

**Review by gcornhill — APPROVED** [review]
🌟 Thank you!

**Pull request merged**